src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs

```diff
@@ -216,6 +437,7 @@ public void Dispose()
                 s_allMeters.Remove(this);
                 instruments = _instruments;
                 _instruments = new List<Instrument>();
+                _nonObservableInstrumentsCache = new();
```

Comment:
I address _nonObservableInstrumentsCache  here but didn't change _instruments because this can be a breaking change. We can consider it in a separate PR.